packages/itwin/tree-widget/src/test/trees/VisibilityTreeEventHandler.test.ts

```diff
@@ -6,26 +6,35 @@
 import { expect } from "chai";
 import { EMPTY, from, Subject } from "rxjs";
 import sinon from "sinon";
-import * as moq from "typemoq";
 import { BeEvent, BeUiEvent, using } from "@itwin/core-bentley";
 import { CheckBoxState } from "@itwin/core-react";
 import { VisibilityTreeEventHandler } from "../../components/trees/VisibilityTreeEventHandler";
 import { flushAsyncOperations } from "../TestUtils";
-import { createSimpleTreeModelNode } from "./Common";
+import { createCategoryNode, createElementNode, createModelNode, createSimpleTreeModelNode, createSubjectNode } from "./Common";
 
 import type { AbstractTreeNodeLoaderWithProvider, CheckboxStateChange, TreeModel, TreeModelChanges, TreeModelSource } from "@itwin/components-react";
 import type { IPresentationTreeDataProvider } from "@itwin/presentation-components";
 import type { SelectionHandler } from "@itwin/presentation-frontend";
 import type { IVisibilityHandler, VisibilityChangeListener, VisibilityStatus, VisibilityTreeEventHandlerParams } from "../../components/trees/VisibilityTreeEventHandler";
+import { IModelApp } from "@itwin/core-frontend";
 
 describe("VisibilityTreeEventHandler", () => {
-  const modelSourceMock = moq.Mock.ofType<TreeModelSource>();
-  const modelMock = moq.Mock.ofType<TreeModel>();
-  const nodeLoaderMock = moq.Mock.ofType<AbstractTreeNodeLoaderWithProvider<IPresentationTreeDataProvider>>();
-  const dataProviderMock = moq.Mock.ofType<IPresentationTreeDataProvider>();
-  const selectionHandlerMock = moq.Mock.ofType<SelectionHandler>();
+  const modelStub = { getNode: () => createSimpleTreeModelNode(), iterateTreeModelNodes: () => {} } as any as TreeModel;
+  const modelSourceStub = {
```

Comment:
Maybe it would be simpler to use real `TreeModelSource` implementation  isntead of stubing `model` and `modelSource`.